### PR TITLE
Backend/company query

### PR DESCRIPTION
**Pull request by Lewin-B**
reference pr 99

### CODE REVIEW

File: client/src/components/ExperienceQuerySearchInput.js [hunk 1]
```diff
@@ -20,14 +20,23 @@ export const ExperienceQuerySearchInput = ({ value, handleSearch }) => {
 
     const handleInputChange = async (event) => {
         const inputValue = event.target.value
-        setQuery(inputValue)
+        console.log('Input Value: ', inputValue)
```
Comment (arfpog) [inline client/src/components/ExperienceQuerySearchInput.js]: remove this

File: client/src/components/ExperienceQuerySearchInput.js [hunk 2]
```diff
@@ -37,10 +46,11 @@ export const ExperienceQuerySearchInput = ({ value, handleSearch }) => {
 
     const handleCompanyButtonClick = async (company) => {
         // reset text input
+        console.log('company: ', company)
```
Comment (arfpog) [inline client/src/components/ExperienceQuerySearchInput.js]: remove this

File: client/src/pages/EditProfile.js [hunk 3]
```diff
@@ -591,6 +592,8 @@ function EditProfile() {
             pipeline,
         }
 
+        console.log('pipeline: ', pipeline)
```
Comment (arfpog) [inline client/src/pages/EditProfile.js]: remove

File: client/src/pages/EditProfile.js [hunk 4]
```diff
@@ -626,6 +629,8 @@ function EditProfile() {
             console.log(response.status)
         }
 
+        console.log('profile: ', profile)
```
Comment (arfpog) [inline client/src/pages/EditProfile.js]: remove

File: server/controllers/companyController.js [hunk 5]
```diff
@@ -57,6 +57,23 @@ const getCompany = async (req, res) => {
...
+
+  // Create a regular expression to find documents that start with the query
+  const regex = new RegExp("^" + query, "i"); // "i" for case-insensitive search
+
+  // Find companies that match the regex
+  const companies = await Company.find({ name: regex });
+
+  res.status(200).json(companies);
+
+  try {
```
Comment (arfpog) [inline server/controllers/companyController.js]: why an empty try catch..?

File: client/src/pages/EditProfile.js (hunk 4 above)
Comment (Lewin-B) [reply client/src/pages/EditProfile.js]: Yes sir

File: client/src/components/ExperienceQuerySearchInput.js (hunk 1 above)
Comment (Lewin-B) [reply client/src/components/ExperienceQuerySearchInput.js]: Yes sir

File: client/src/components/ExperienceQuerySearchInput.js (hunk 2 above)
Comment (Lewin-B) [reply client/src/components/ExperienceQuerySearchInput.js]: Yes sir

File: client/src/pages/EditProfile.js (hunk 3 above)
Comment (Lewin-B) [reply client/src/pages/EditProfile.js]: Yes sir

File: server/controllers/companyController.js (hunk 5 above)
Comment (Lewin-B) [reply server/controllers/companyController.js]: I was supposed to put the code in there lmao my bad

**Pull request merged**